hoomd/md/ForceComposite.h

```diff
@@ -73,6 +73,49 @@ class PYBIND11_EXPORT ForceComposite : public MolecularForceCompute
          */
         virtual void validateRigidBodies(bool create=false);
 
+        
+        /// Construct from a Python dictionary
+        void setBody(std::string typ, pybind11::object v)
+            {   
+            pybind11::list types = v["types"];
+            pybind11::list positions = v["positions"];
+            int N = (unsigned int)len(positions);
+
+            // extract the positions from the python list
+            std::vector<vec3<Scalar>> pos_vector;
+            for (unsigned int i = 0; i < N; i++)
+                {   
+                pybind11::list pos_i = positions[i];
+                if (len(pos_i) != 3)
+                    throw std::runtime_error("Each position must have 3 coordinates");
+                vec3<Scalar> pos = vec3<Scalar>(pybind11::cast<Scalar>(pos_i[0]),
+                                                pybind11::cast<Scalar>(pos_i[1]),
+                                                pybind11::cast<Scalar>(pos_i[2]));
+                pos_vector.push_back(pos);
+                }   
+
+            }   
+
+        /// Convert parameters to a python dictionary
+        pybind11::dict getBody(std::string typ)
+            {
+            pybind11::dict v;
+            pybind11::list types;
+            pybind11::list positions;
+            for(unsigned int i = 0; i < N; i++)
```

Comment:
OK, I know I am doing something wrong, but I don't understand how variables are moving between python and C. I am trying to get the properties of (some C object I think?) set by `setBody` N should be the length of the positions and x, y, and z should be the coordinates. How to I access those correctly?

Reply:
`setBody` moves values from the Python dictionary the user provides and stores them in the internal C++ data structures that `ForceComposite` manges.

`getBody` pulls them out of the internal C++ data structures and packs them in a dictionary.

The data structures are dense arrays that store the per-type-per-constituent-particle data. Get the number of particles in a given body with:
```
ArrayHandle<unsigned int> h_body_len(m_body_len, access_location::host, access_mode::readwrite);
unsigned int N h_body_len.data[body_type_id];
```

To get the position of particle `i` in this body and pack it in a list, use:
```
ArrayHandle<Scalar3> h_body_pos(m_body_pos, access_location::host, access_mode::read);
pybind11::list pos;
pos.append(h_body_pos.data[m_body_idx(body_typeid,i)].x);
pos.append(h_body_pos.data[m_body_idx(body_typeid,i)].y);
pos.append(h_body_pos.data[m_body_idx(body_typeid,i)].z);
```

See the contents of `setParam` for details.